src/pages/clubs/index.js

```diff
@@ -118,57 +118,64 @@ class Clubs extends React.PureComponent {
     }
   `;
 
-  // function
-  // Updates the state when new data  is fetched.
+  // Updates the state when new data is fetched.
   onClubsFetched = data => {
     const snapshot = { ...this.state };
     const shownClubs = [...snapshot.shownClubs, ...data.clubs.clubs];
-
     this.setState(() => ({
       shownClubs: shownClubs,
       cursor: data.clubs.pageInfo.endCursor,
-      firstLoad: false,
+      firstLoad: snapshot.view === 'map' && data.clubs.pageInfo.hasNextPage,
       hasNextPage: data.clubs.pageInfo.hasNextPage
     }));
   };
 
+  // loads the first data.
+  clubQuery = snapshot => {
+    let content;
+
+    content = (
+      <Query
+        query={this.GET_CLUBS}
+        variables={{
+          number: snapshot.number,
+          cursor: snapshot.cursor,
+          search: snapshot.searchString === '' ? null : snapshot.searchString
+        }}
+        onCompleted={data => {
+          this.onClubsFetched(data);
+        }}
+      >
+        {({ loading, error }) => {
+          if (loading) {
+            return (
+              <div className="text text--large text--load-more">
+                Loading....
+              </div>
+            );
+          }
+          if (error) {
+            return <div> `Error ${error.message}` </div>;
+          } else {
+            // JSX elements
+            // create the DOM for the component.
+            return null;
+          }
+        }}
+      </Query>
+    );
+
+    return content;
+  };
+
   // function
   // loads the first data.
   onFirstLoad = () => {
     const snapshot = { ...this.state };
     let content;
 
     if (snapshot.firstLoad) {
```

Comment:
The if here is not needed, just do `return this.state.firstLoad ? this.clubQuery() : ""`